scri/asymptotic_bondi_data/map_to_superrest_frame.py

```diff
@@ -404,22 +404,31 @@ def com_transformation_to_map_to_superrest_frame(abd, N_itr_max=10, rel_err_tol=
     return best_CoM_transformation, rel_errs
 
 
-def rotation_from_spin_charge(chi, t):
+def rotation_from_spin_charge(chi, t, fix_xz_plane=False):
     """Obtain the rotation from the remnant BH's spin vector.
-    This finds the rotation that aligns the z-component of the spin vector with the z-axis.
+
+    This finds the rotation that aligns the z-component of the spin vector with the z-axis,
+    i.e., this finds the quaternion q such that q * quaternion.z * q.inverse() = chi_f(t).
+    
     Parameters
     ----------
     chi: ndarray, real, shape (..., 3)
         Remnant BH's spin vector.
     t: ndarray, real
         Time array corresponding to the size of the spin vector.
+    fix_xz_plane: bool
+        Whether or not to modify the quaternion by a rotation about z
+        such that the x unit vector remains in the x-z plane
+        after the full rotation.
+        Default is False.
     """
     chi_f = quaternion.quaternion(*chi[np.argmin(abs(t))]).normalized()
     q = (1 - chi_f * quaternion.z).normalized()
 
-    x_rotated = quaternion.as_vector_part(q.inverse() * quaternion.x * q)
-    phase = np.angle(x_rotated[0] + 1j*x_rotated[1])
-    q = q * quaternion.from_rotation_vector(phase * np.array([0, 0, 1]))
+    if fix_xz_plane:
+        y_rotated = quaternion.as_vector_part(q * quaternion.y * q.inverse())
+        phase = np.angle(1j*(y_rotated[0] + 1j*y_rotated[1]))
+        q = q * quaternion.from_rotation_vector(phase * np.array([0, 0, 1]))
```

Comment:
Something's gone wrong here: if `chi = z`, I would expect to get `q=1` at the end, but this gives me `q=z`.

I think it should be `phase = np.atan2(-y_rotated[0], y_rotated[1])`.

Reply:
Ah yeah good catch; `np.angle(-1j*(y_rotated[0] + 1j*y_rotated[1]))` works too, but using `atan2` is fine by me!